aiohttp_devtools/runserver/watch.py

```diff
@@ -119,10 +121,33 @@ def _start_dev_server(self) -> None:
         self._process = Process(target=serve_main_app, args=(self._config, tty_path))
         self._process.start()
 
-    def _stop_dev_server(self) -> None:
+    async def _stop_dev_server(self) -> None:
         if self._process.is_alive():
             logger.debug('stopping server process...')
+            if self._config.shutdown_endpoint:  # a workaround for singals not working on Windows
+                url = 'http://localhost:{}{}/shutdown'.format(self._config.main_port, self._config.path_prefix)
+                logger.debug('attempting to stop process via shutdown endpoint {}'.format(url))
+                try:
+                    # these errors are expected because the request kills the server *immediately*
+                    with suppress(ServerDisconnectedError, ClientConnectorError):
+                        async with ClientSession() as session:
+                            async with session.get(url):
+                                pass
+                except (ConnectionError, ClientError, asyncio.TimeoutError) as ex:
+                    if self._process.is_alive():
+                        logger.warning("shutdown endpoint caused an error (will try signals next): {}".format(ex))
+                    else:
+                        logger.warning("process stopped (despite error at shutdown endpoint: {})".format(ex))
+                        return
+                else:
+                    self._process.join(5)
```

Comment:
This is needed because SIGINT will kill immediately on Windows, right?

Reply:
My thought was that if the shutdown endpoint doesn't work for whatever reason, it'll fall back to SIGINT, which does immediately kill the process, without the graceful shutdown.